script/GameObjects/Components/Core/Health.lua

```diff
@@ -12,10 +12,26 @@ end
 
 function Entity:damage (amount, source)
   assert(self.health)
+
   if self.health <= 0 then return end
+
   self.health = max(0, self.health - amount)
   self:send(Event.Damaged(amount, source))
-  if self.health <= 0 then
+
+  if self.health <= 0.009999999 then
```

Comment:
what is that lol :D

Reply:
It's catching the case where health has been reduced to within two decimal places of zero. A test for exactly zero might not work as intended.

Reply:
<= tests for below and exactly 0 tho, should be fine for most cases ;)

Reply:
you can use math.floor to round health to 2 decimals If needed

Reply:
Code will be changed to use floor() in next commit.